Helpers/HttpExceptionFactory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Exceptions\Helpers;
+
+use Exceptions\Http\Client;
+use Exceptions\Http\HttpExceptionInterface;
+use Exceptions\Http\Server;
+use InvalidArgumentException;
+use Throwable;
+
+class HttpExceptionFactory
```

Comment:
I think there is some value to this implementation but what i'm scared is it creates a loose coupling in your code. Let's assume this here: I have a piece of code that should throw a specific 404 exception, but because i don't want to type out the exception's name i'll type the factory name instead and use the buildException to pass in 404 error code. Seems counter intuitive to me, why not just send back the real exception in the first place anyway.

This means that the user can now throw the exception 2 different ways, one in a loosely coupled way and one with a tightly coupled way. 

The first problem this causes is it makes your IDE wonder exactly what you are throwing. It knows you are throwing an HttpExceptionInterface but doesn't know which one which creates some kind of covariance problem. This isn't super critical in my mind but it's worth noting.

The second problem i see is if someone wants to inspect where they are using NotFoundException in their app, they now have 2 ways to search. It's not that bad but i dislike useless forks in the code. I try to keep the code i build pure with only one approach. This also applies to docblocks, if you use the factory, you can't just specific HttpExceptionInterface in the docblock because it dumbs down your type hinting and documentation which makes it less readable.